std/algorithm/searching.d

```diff
@@ -1466,6 +1476,25 @@ private auto extremum(alias selector = "a < b", Range,
     assert(arr2d.extremum!"a[1]" == arr2d[1]);
 }
 
+// https://issues.dlang.org/show_bug.cgi?id=17982
+@safe unittest
+{
+    class B
+    {
+        int val;
+        this(int val){ this.val = val; }
+    }
+
+    const(B) doStuff(const(B)[] v)
+    {
+        return v.extremum!"a.val";
+    }
+
+    const(B)[] arr = [new B(0), new B(1)];
+    // can't compare directly - https://issues.dlang.org/show_bug.cgi?id=1824
+    assert(arr.extremum!"a.val".val == 0);
```

Comment:
Imho this is the real bummer - `arr.extremum!"a.val" == new B(0)` doesn't work :/